src/resolve/resolved-dns-cache.c

```diff
@@ -1265,6 +1266,14 @@ int dns_cache_export_shared_to_packet(DnsCache *cache, DnsPacket *p) {
                         if (!j->shared_owner)
                                 continue;
 
+                        if (t <= 0)
+                                t = now(CLOCK_BOOTTIME);
+
+                        /* RFC6762 7.1: Don't append records with less than
+                         * half the TTL remaining as known answers. */
+                        if ((t >= j->until) || (j->until - t) < (0.5 * (j->rr->ttl * USEC_PER_SEC)))
```

Comment:
I doubt thing this needs float math:
```suggestion
                        if ((t >= j->until) || (j->until - t) < (j->rr->ttl * (USEC_PER_SEC / 2)))
```

Reply:
Changed as suggested